### DISCUSSION TITLE
Backward compatibility between Joomla 3 and Joomla 4 URL format?

### DISCUSSION THREAD

**Discussion by PhocaCz** [joomla/joomla-cms#37722] [Q&A]
Hi, 

I haven't dealt with this topic yet, so if I'm missing some basic information to solve this problem, or it's been discussed a lot somewhere, I apologize for the duplicate questions.

I get a lot of questions about how to keep the same links when migrating from J3 to J4.

In J3, links have IDs for articles and categories (if a category has subcategory then they share one ID).

J3:
alias/10-category/subcategory/1-article

J4:
alias/category/subcategory/article
alias/9-category/10-subcategory/1-article

In J4, links have no IDs or if the ID feature is enabled, they have IDs for each part (the difference is the category and subcategory part where each category in URL has own ID).

Is there some solution in J4 to have the same URL format like in J3? Because I can imagine, changing almost all URLs on the website is not good practice when migrating.

Thank you, Jan

**Comment by Fedik** [comment]
One of solution I can think is to use Redirect component.
alias/10-category/subcategory/1-article => alias/category/subcategory/article

If there a loot of content, then wait when 404 will be collected, and then set redirect.
Not perfect, but should work.

Other idea: write a plugin that will catch 404, and do check if it "old url".

**Comment by PhocaCz** [comment]
Yes, but I hoped we could avoid the redirect. When ID is enabled, there is really small difference:

J3:
alias/10-category/subcategory/1-article

J4:
alias/9-category/10-subcategory/1-article

(missing ID for parent category in J3)
